app/lib/features/chat/widgets/create_chat.dart

```diff
@@ -448,6 +248,217 @@ class _CreateChatWidgetConsumerState extends ConsumerState<_CreateChatWidget> {
+      replacement: const SizedBox.shrink(),
+      child: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: Wrap(
+          direction: Axis.horizontal,
+          spacing: 5.0,
+          runSpacing: 5.0,
+          children: List.generate(
+            selectedUsers.length,
+            (index) => Container(
+              padding: const EdgeInsets.all(8),
+              decoration: BoxDecoration(
+                borderRadius: BorderRadius.circular(30),
+                border: Border.all(
+                  color: Theme.of(context).colorScheme.onSurface,
+                ),
+              ),
+              child: Consumer(
+                builder: (context, ref, child) {
+                  final avatarProv =
+                      ref.watch(userAvatarProvider(selectedUsers[index]));
+                  final displayName = selectedUsers[index].getDisplayName();
+                  final userId = selectedUsers[index].userId().toString();
+                  return Row(
+                    mainAxisSize: MainAxisSize.min,
+                    children: <Widget>[
+                      ActerAvatar(
+                        mode: DisplayMode.DM,
+                        avatarInfo: AvatarInfo(
+                          uniqueId: userId,
+                          displayName: displayName ?? userId,
+                          avatar: avatarProv.valueOrNull,
+                        ),
+                        size: 14,
+                      ),
+                      const SizedBox(width: 5),
+                      Text(
+                        displayName ?? userId,
+                        style: Theme.of(context)
+                            .textTheme
+                            .labelMedium!
+                            .copyWith(
+                              color: Theme.of(context).colorScheme.onSurface,
+                            ),
+                      ),
+                      const SizedBox(width: 10),
+                      InkWell(
+                        onTap: () =>
+                            ref.read(_selectedUsersProvider.notifier).update(
+                                  (state) => [
+                                    for (int j = 0; j < state.length; j++)
+                                      if (j != index) state[j],
+                                  ],
+                                ),
+                        child: Icon(
+                          Icons.close_outlined,
+                          size: 18,
+                          color: Theme.of(context).colorScheme.onSurface,
+                        ),
+                      ),
+                    ],
+                  );
+                },
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget renderPrimaryAction(BuildContext context) {
+    final selectedUsers = ref.watch(_selectedUsersProvider).toList();
+    return ListTile(
+      onTap: selectedUsers.isEmpty
```

Comment:
I think we can also improve this tap handler having separate ```handleCreateAction()``` for readability.

Reply:
next time ;) .